### ISSUE TITLE
Document what's new in ASP.NET Core for .NET 9 Preview 1

### ISSUE THREAD

**Issue by danroth27** [dotnet/AspNetCore.Docs#31699]
Update the "What's new in ASP.NET Core 9.0" for .NET 9 Preview 1. Creating the doc is tracked by #31698.

@adityamandaleeka @mkArtakMSFT Please comment on this issue with the desired content.

FYI: @samsp-msft @JeremyLikness @mikekistler @claudiaregio @JamesNK @SteveSandersonMS @davidfowl

**Comment by Rick-Anderson** [comment]
Page: https://learn.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-9.0
MD file: https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/release-notes/aspnetcore-9.0.md
Doc ID: 4e75ad25-2c3f-b28e-6a91-ac79a9c683b6

**Comment by JamesNK** [comment]
In .NET 9 Preview 1 we've updated the debugging display of dictionaries and other name/value collections to use an improved layout. The name is now displayed in the debugger's name column instead of being concatenated with the value.

You can see an example of the new debugging display in the example with a dictionary:

![GBmvgiubYAAOsk7](https://github.com/dotnet/AspNetCore.Docs/assets/303201/35cc7978-eb41-49c6-9486-aac78446fa99)

ASP.NET Core has a lot of name/values. Improved debugging is applied everywhere:

* HTTP headers
* Query strings
* Forms
* Cookies
* View data
* Route data
* Features
* And more

**Comment by claudiaregio** [comment]
@danroth27 Thanks for putting this together! Nothing in my areas yet

**Comment by danroth27** [comment]
Nothing new for Blazor/MVC/Razor Pages as well. We just did bug fixes.

**Comment by BrennanConroy** [comment]
## Json Polymorphic type support in SignalR

Hub methods can now accept a base class instead of the derived class to enable polymorphic scenarios. The base type needs to be annotated to allow polymorphism, details are available at https://learn.microsoft.com/dotnet/standard/serialization/system-text-json/polymorphism?pivots=dotnet-8-0

```
    public class MyHub : Hub
    {
        public void Method(JsonPerson person)
        {
            if (person is JsonPersonExtended)
            {
            }
            else if (person is JsonPersonExtended2)
            {
            }
            else
            {
            }
        }
    }

    [JsonPolymorphic]
    [JsonDerivedType(typeof(JsonPersonExtended), nameof(JsonPersonExtended))]
    [JsonDerivedType(typeof(JsonPersonExtended2), nameof(JsonPersonExtended2))]
    private class JsonPerson
    {
        public string Name { get; set; }
        public Person Child { get; set; }
        public Person Parent { get; set; }
    }

    private class JsonPersonExtended : JsonPerson
    {
        public int Age { get; set; }
    }

    private class JsonPersonExtended2 : JsonPerson
    {
        public string Location { get; set; }
    }
```

**Comment by baktay** [comment]
@danroth27 There is an important issue I came across.

I have created a website using Blazor Server Web App in .NET8. In my project I have dynamically created SEO tags. For example, dynamic meta description that changes according to the page. I also have a canonical tag telling search crawlers the uri (automatically calculated by razor page code section as navigation changes). 

I discovered that google search console did not see any of these tags on my website. When I inspected my page from browser devtools, I saw all the tags in Elements Page, however the Page Source does not have any of these tags. Not even my <h1> tag in the HTML body which is essential for SEO. Is this a known issue? The app.razor page seems to have script tags correctly but I could not create dynamic tags for SEO (scripts in HTML head tag per page, for example).

When I inspect my page from Google Search Console, the crawler bot seems to grab a similar page the Page Source code, and not the Element Page where the HTML tags are visible. 

Is .NET9 or any .NET8 updates going to solve the issue? If not, I think Blazor Server is not for web design as it will have no visible SEO tags for crawler bots. 

I also inspected my site with webmaster tools at Bing. It is not reading my <h1> tags either. I don't want to go back to previous versions of asp.net since I enjoy the benefits of Blazor, but SEO is very important. I have not tried client-side WASM, but I like server-side. I did spend quite some time on this. I hope I am not missing something major. Thank You if you have taken the time to read this message.
